.github/actions/install-whitaker/tests/test_install_whitaker.py

```
@@ -0,0 +1,724 @@
+        assert "title=Whitaker installer::status=complete" in install_script
+        assert "GITHUB_STEP_SUMMARY" in install_script
+        assert 'CARGO_HOME="${CARGO_HOME:-$HOME/.cargo}"' in install_script
+
+    def test_normalizes_valid_action_inputs(self, tmp_path: Path) -> None:
+        """Verify validation expands the supported tilde Cargo-home form."""
+        result = _run_input_validation(tmp_path, "~/.cargo", "1.2.3")
+
+        assert result.returncode == 0, result.stderr
+        assert (tmp_path / "output").read_text(encoding="utf-8").splitlines() == [
+            "cargo-home="
+            f"{_bash_path(shutil.which('bash') or 'bash', tmp_path / 'home')}/.cargo",
```

Ruff (0.16.1)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)